### DISCUSSION TITLE
[Tree] Slot for icons

### DISCUSSION THREAD

**Discussion by ig-onoffice-de** [primefaces/community#414] [PrimeVue]
I would like to have slots for the Icons in the Tree component. We use Font Awesome icons and want to use the `FontAwesomeIcon` component.

**Comment by FlipWarthog** [comment]
There are four different slots for icons in Tree: https://primevue.org/tree/#api.tree.slots

**Reply by ig-onoffice-de** [reply]
Yes, but not for the tree-icon.

**Reply by FlipWarthog** [reply]
I'm not sure what you mean, can you post a screenshot of what you're referring to?

**Reply by ig-onoffice-de** [reply]
![2023_11_09_16_14_00_Vue_Tree_Component](https://github.com/primefaces/community/assets/85568061/4bb3c1be-0122-42b3-b66a-493aa70eb708)

**Reply by FlipWarthog** [reply]
I see, so that would be the `icon` value in TreeNode, but that's just a String for icon classes that get applied to a `span`: https://primevue.org/tree/#api.tree.interfaces.TreeNode.icon

Are you able to use that with the Font Awesome icon classes themselves, e.g., `icon: 'fas fa-envelope'`?

**Reply by ig-onoffice-de** [reply]
No, I'm not. I don't want to import all the icons. I want to use the [FontAwesomeIcon component](https://fontawesome.com/docs/web/use-with/vue/dig-deeper#computed-property) and import icon by icon as computed property.

**Reply by FlipWarthog** [reply]
I'm not sure if it's feasible, but you might open an issue in the PrimeVue repo and see what PrimeTek says.

**Reply by ig-onoffice-de** [reply]
I wanted to do that, but they led me here for feature requests 😅

**Reply by ig-onoffice-de** [reply]
Created this [issue](https://github.com/primefaces/primevue/issues/4833)

**Reply by FlipWarthog** [reply]
Ah got it. I think PrimeTek has an internal tracker for potential enhancements. cc. @tugcekucukoglu for confirmation

**Comment by tugcekucukoglu** [comment]
Yes, we have an internal tracker for improvements and/or new features. We also think that discussions are the most appropriate place for users' feedback. Comments from the community in edge cases will also speed up the process. We want to use the issue tracker for potential and/or existing open bugs. We first started this process at PrimeVue, and the PrimeReact and PrimeNG teams will enter a similar process.

To talk about this issue, the icon slot can always be added, but here it is necessary to decide whether the icon option given in the node should override the item slot. Since this is not a bug, it should be discussed before opening an issue.

**Reply by ig-onoffice-de** [reply]
The icon option of the node should be passed to the slot (or the whole node) - should be the same as in default slot.

**Reply by ig-onoffice-de** [reply]
Would it be worth it to open a PR with the changes? I really need this slot 😅
